pkg/util/net/http.go

```diff
@@ -120,3 +120,28 @@ func GetHTTPClient(req *http.Request) string {
 	}
 	return "unknown"
 }
+
+// Extracts and returns the clients IP from the given request.
+func GetClientIP(req *http.Request) (net.IP, error) {
+	hdr := req.Header
+	// First check the X-Forwarded-For header for requests via proxy.
+	hdrForwardedFor := hdr.Get("X-Forwarded-For")
+	if hdrForwardedFor != "" {
+		// X-Forwarded-For can be a csv of IPs in case of multiple proxies.
+		// Use the first one.
+		parts := strings.Split(hdrForwardedFor, ",")
+		ip := net.ParseIP(strings.TrimSpace(parts[0]))
```

Comment:
Added test.
Also updated the func comment to explicitly say that it can return nil.
The current code in genericserver handles it fine. cidr.Matches() can handle nil.